.ci/no_old_refs.sh

```diff
@@ -3,29 +3,76 @@
 set -e
 
 # Script requires GITHUB_TOKEN env variable
-MENTIONED_ISSUES=/tmp/mentioned_issues
+MENTIONED_ISSUES_GREP_OUTPUT=/tmp/mentioned_issues_grep_output
 CLOSED_ISSUES=/tmp/failed_issues
+LINKED_ISSUES_FORMATTED=/tmp/linked_issues
+CHECKSTYLE_ISSUE_URL="https:\/\/github.com\/checkstyle\/checkstyle\/issues\/"
+
+# $LINKED_ISSUES need formatting before the are used
+if [ ! -z "$LINKED_ISSUES" ]; then
+  echo $LINKED_ISSUES | sed -e 's/,/\n/g' >> $LINKED_ISSUES_FORMATTED
+  sed -i "s/^/$CHECKSTYLE_ISSUE_URL/" $LINKED_ISSUES_FORMATTED
+fi
+
+# Linked issues that are mentioned in the code
+LINKED_ISSUES_MENTIONED=/tmp/linked_issues_mentioned
 API_GITHUB_PREFIX="https://api.github.com/repos"
 GITHUB_HOST="https://github.com"
+MAIN_REPO="checkstyle/checkstyle"
+DEFAULT_BRANCH="master"
+
+# These are modified when event is of type pull_request
+if [ ! -z "$PR_HEAD_REPO_NAME" ]; then
+  MAIN_REPO=$PR_HEAD_REPO_NAME
+  DEFAULT_BRANCH=$GITHUB_HEAD_REF
+fi
 
 # collect issues where full link is used
-grep -IPohr "(after|[Tt]il[l]?) $GITHUB_HOST/[\w.-]+/[\w.-]+/issues/\d{1,5}" . \
-  | sed -e 's/.*github.com\///' >> $MENTIONED_ISSUES
+grep -IPonr "(after|[Tt]il[l]?) $GITHUB_HOST/[\w.-]+/[\w.-]+/issues/\d{1,5}" . \
+  | sed -e 's/:[a-zA-Z].*github.com\//:/' >> $MENTIONED_ISSUES_GREP_OUTPUT
 
 # collect checkstyle issues where only hash sign is used
-grep -IPohr "[Tt]il[l]? #\d{1,5}" . \
-  | sed -e 's/.*#/checkstyle\/checkstyle\/issues\//' >> $MENTIONED_ISSUES
+grep -IPonr "[Tt]il[l]? #\d{1,5}" . \
+  | sed -e 's/:[a-zA-Z].*#/:checkstyle\/checkstyle\/issues\//' >> $MENTIONED_ISSUES_GREP_OUTPUT
+
+for MENTIONED_ISSUES_GREP_OUTPUT_LINE in $(cat $MENTIONED_ISSUES_GREP_OUTPUT); do
+  ISSUE=${MENTIONED_ISSUES_GREP_OUTPUT_LINE#*[0-9]:}
+
+  FILE_PATH=${MENTIONED_ISSUES_GREP_OUTPUT_LINE%:[0-9]*}
+  FILE_PATH=${FILE_PATH:2}
 
-for issue in $(sort -u $MENTIONED_ISSUES); do
-  STATE=$(curl -s -H "Authorization: token $GITHUB_TOKEN" "$API_GITHUB_PREFIX/$issue" \
+  LINE_NUMBER=${MENTIONED_ISSUES_GREP_OUTPUT_LINE#*:}
+  LINE_NUMBER=${LINE_NUMBER%:*}
+
+  LINK="$GITHUB_HOST/$MAIN_REPO/blob/$DEFAULT_BRANCH/$FILE_PATH#L$LINE_NUMBER"
+
+  STATE=$(curl -s -H "Authorization: token $GITHUB_TOKEN" "$API_GITHUB_PREFIX/$ISSUE" \
    | jq '.state' | xargs)
   if [ "$STATE" = "closed" ]; then
-    echo "$GITHUB_HOST/$issue" >> $CLOSED_ISSUES
+    echo "$LINK" >> $CLOSED_ISSUES
+  fi
+  if [ ! -z "$LINKED_ISSUES" ]; then
+    for LINKED_ISSUE in $(cat $LINKED_ISSUES_FORMATTED); do
+      if [ "$LINKED_ISSUE" = "$GITHUB_HOST/$ISSUE" ]; then
+        echo "$LINK" >> $LINKED_ISSUES_MENTIONED
+      fi
+    done
   fi
 done
 
+EXIT_CODE=0
+
 if [ -f "$CLOSED_ISSUES" ]; then
-    echo "Following issues are mentioned in code to do something after they are closed:"
+    echo -ne "\nFollowing lines are referencing closed issues, such lines should be removed:\n\n"
```

Comment:
`such lines should be removed` that is misleading, may be issue should be reopened or new one has to be created.

Reply:
I modified the message, please take a look.

Reply:
wording lgtm now, but please split it to 2 sentences and 2 echos. This will also fix my comment https://github.com/checkstyle/checkstyle/pull/11443#discussion_r839979595

Reply:
Done